pkg/storaged/nfs-mounts.py

```diff
@@ -0,0 +1,344 @@
+        self.inotify.process(event)
+
+def field_escape(str):
+    return str.replace("\\", "\\134").replace(" ", "\\040").replace("\t", "\\011")
+
+def field_unescape(str):
+    return re.sub("\\\\([0-7]{1,3})", lambda m: chr(int(m.group(1), 8)), str)
+
+def parse_tab(name):
+    entries = [ ]
+    for line in open(name, "r").read().split("\n"):
+        sline = line.strip()
+        if sline == "" or sline[0] == "#":
+            continue
+        fields = list(map(field_unescape, re.split("[ \t]+", sline)))
+        if len(fields) > 0 and ":" in fields[0]:
+            entries.append(fields)
+    return entries
+
+def index_tab(tab):
+    by_remote = { }
+    for t in tab:
+        if not t[0] in by_remote:
+            by_remote[t[0]] = [ ]
+        by_remote[t[0]].append(t)
+    return by_remote
+
+def modify_tab(name, modify):
+    lines = open(name, "r").read().split("\n")
+    if len(lines) > 0 and lines[len(lines)-1] == "":
+        del lines[len(lines)-1]
+
+    new_lines = [ ]
+    for line in lines:
+        sline = line.strip()
+        if sline == "" or sline[0] == "#":
+            new_lines.append(line)
+        else:
+            fields = list(map(field_unescape, re.split("[ \t]+", sline)))
+            if len(fields) > 0 and ":" in fields[0]:
+                new_fields = modify(fields)
+                if new_fields:
+                    if new_fields == fields:
+                        new_lines.append(line)
+                    else:
+                        new_lines.append(" ".join(map(field_escape, new_fields)))
+            else:
+                new_lines.append(line)
+    new_fields = modify(None)
+    if new_fields:
+        new_lines.append(" ".join(map(field_escape, new_fields)))
+
+    f = open(name + ".tmp", "w")
+    f.write("\n".join(new_lines) + "\n")
+    f.flush()
+    os.fsync(f.fileno())
+    f.close()
+    os.rename(name + ".tmp", name)
+
+fstab = [ ]
+fstab_by_remote = { }
+
+mtab = [ ]
+mtab_by_remote = { }
+
+def process_fstab():
+    global fstab, fstab_by_remote
+    fstab = parse_tab("/etc/fstab")
+    fstab_by_remote = index_tab(fstab)
+
+def process_mtab():
+    global mtab, mtab_by_remote
+    mtab = parse_tab("/etc/mtab")
+    mtab_by_remote = index_tab(mtab)
+
+def find_in_tab(tab_by_remote, fields):
+    for t in tab_by_remote.get(fields[0], [ ]):
+        if t[0] == fields[0] and t[1] == fields[1]:
+            return t
+    return None
+
+def report():
+    data = [ ]
+    for f in fstab:
+        m = find_in_tab(mtab_by_remote, f)
+        data.append({ "fstab": True, "fields": f, "mounted": m is not None })
+    for m in mtab:
+        if not find_in_tab(fstab_by_remote, m):
+            data.append({ "fstab": False, "fields": m, "mounted": True })
+    sys.stdout.write(json.dumps(data) + "\n")
+    sys.stdout.flush()
+
+def monitor():
+    process_mtab()
+    process_fstab()
+    report()
+    fstab_watcher = Watcher("/etc/fstab")
+    mtab_file = open("/proc/mounts", "r")
```

Comment:
Let's be consistent and use `/proc/self/mounts` here, too. Like `/etc/mtab` itself, `/proc/mounts` is just a symlink and thus rather useless to monitor.

Reply:
Yes.